lib/handler/range.c

```diff
@@ -0,0 +1,112 @@
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "h2o.h"
+
+
+#define CHECK_EOF()                             \
+    if (buf == buf_end) {                       \
+        *ret = -2;                              \
+        return NULL;                            \
+    }
+
+#define EXPECT_CHAR(ch)                         \
+    CHECK_EOF();                                \
+    if (*buf++ != ch) {                         \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+#define CHECK_OVERFLOW(range)                   \
+    if (range == SIZE_MAX) {                    \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+size_t *process_range(h2o_mem_pool_t *pool, h2o_iovec_t *range_value, size_t file_size, size_t* ret)
```

Comment:
The function seems to fail in parsing the following patterns:
* `bytes=1-3 5-10` (IMO should be considered invalid per spec)
* `bytes=1-\t,5-10` (IMO should return two ranges)

To me it seems like that it is possible to fix the issues by making the logic simpler by doing the following:
* logic that skips `[ \t]*,[ \t]*` should exist only at the top of the do-while block
 * current code does some checks at the top, and others within the logic that extracts a single range
* rest of the do-while block should concentrate on extracting a single range, until it reaches EOF or an unknown character